redisdb/datadog_checks/redisdb/redisdb.py

```diff
@@ -131,17 +131,16 @@ def _parse_dict_string(self, string, key, default):
             self.log.exception("Cannot parse dictionary string: %s", string)
             return default
 
-    def _generate_instance_key(self, instance):
-        if 'unix_socket_path' in instance:
-            return instance.get('unix_socket_path'), instance.get('db')
+    @staticmethod
+    def _generate_instance_key(instance_config):
+        if 'unix_socket_path' in instance_config:
+            return instance_config.get('unix_socket_path'), instance_config.get('db')
         else:
-            return instance.get('host'), instance.get('port'), instance.get('db')
+            return instance_config.get('host'), instance_config.get('port'), instance_config.get('db')
 
-    def _get_conn(self, instance=None):
-        if instance is None:
-            instance = self.instance
-        no_cache = is_affirmative(instance.get('disable_connection_cache', False))
-        key = self._generate_instance_key(instance)
+    def _get_conn(self, instance_config):
```

Comment:
Is there a reason to keep `instance_config` argument if `_get_conn` is can get values from `self.instance` directly?

Reply:
Yes. We pass argument other than `self.instance` here:
https://github.com/DataDog/integrations-core/blob/57421b7735bee4abf3af3a1f101775918a14028f/redisdb/datadog_checks/redisdb/redisdb.py#L323